integreat_cms/cms/views/content_revision_view.py

```diff
@@ -0,0 +1,363 @@
+            messages.warning(
+                request,
+                _("You don't have the permission to restore versions of this page."),
+            )
+
+        selected_revision = content_translations.filter(
+            version=kwargs.get("selected_revision", content_translations.count())
+        )
+
+        if not selected_revision.exists():
+            messages.error(request, _("This revision does not exist."))
+            return redirect(
+                f"{content_name}_revisions",
+                **{
+                    f"{content_name}_id": content.id,
+                    "region_slug": region.slug,
+                    "language_slug": language.slug,
+                },
+            )
+
+        if content.archived:
+            messages.warning(
+                request,
+                _("You cannot restore versions of this page because it is archived."),
+            )
+
+        kwargs = {
+            "region_slug": region.slug,
+            "language_slug": language.slug,
+            f"{content_name}_id": content.id,
+        }
+        revision = reverse(f"{content_name}_revisions", kwargs=kwargs)
+        form = reverse(f"edit_{content_name}", kwargs=kwargs)
+
+        return render(
+            request,
+            self.template_name,
+            {
+                **self.get_context(request, *args, **kwargs),
+                "content_name": content_name,
+                "content": content,
+                "content_translations": content_translations,
+                "api_revision": content_translations.filter(
+                    status=status.PUBLIC
+                ).first(),
+                "selected_revision": selected_revision.first(),
+                "language": language,
+                "form": form,
+                "revision": revision,
+            },
+        )
+
+    # pylint: disable=too-many-branches
+    def post(self, request, *args, **kwargs):
+        r"""
+        Restore a previous revision of a page translation
+
+        :param request: The current request
+        :type request: ~django.http.HttpRequest
+
+        :param \*args: The supplied arguments
+        :type \*args: list
+
+        :param \**kwargs: The supplied keyword arguments
+        :type \**kwargs: dict
+
+        :raises ~django.core.exceptions.PermissionDenied: If user does not have the permission to edit the specific page
+
+        :return: The rendered template response
+        :rtype: ~django.template.response.TemplateResponse
+        """
+        content_type = None
+        content_name = ""
+
+        if kwargs.get("page_id"):
+            content_type = Page
+            content_name = "page"
+        elif kwargs.get("poi_id"):
+            content_type = POI
+            content_name = "poi"
+
+        region = request.region
+        content = (
+            content_type.objects.filter(region=region)
+            .filter(id=kwargs.get(f"{content_name}_id"))
+            .first()
+        )
```

Comment:
This query could be handled by the [`SingleObjectMixin`](https://docs.djangoproject.com/en/3.2/ref/class-based-views/mixins-single-object/#singleobjectmixin)